File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/Counter.java
package eu.darken.mvpbakery.example.screens.counting;

public class Counter {

    private int i;

    public Counter(int i) {
        this.i = i;
    }

    void setCounter(int i) {
        this.i = i;
    }

    int countUp() {
        return ++i;
    }

    int getCurrent() {
        return i;
    }

}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/Counter.kt
package eu.darken.mvpbakery.example.screens.counting

class Counter(i: Int) {

    internal var current: Int = 0
        private set

    init {
        this.current = i
    }

    internal fun setCounter(i: Int) {
        this.current = i
    }

    internal fun countUp(): Int {
        return ++current
    }

}
